api/envoy/data/cluster/v3/outlier_detection_event.proto

```diff
@@ -144,3 +150,11 @@ message OutlierEjectFailurePercentage {
   // Host's success rate at the time of the ejection event on a 0-100 range.
   uint32 host_success_rate = 1 [(validate.rules).uint32 = {lte: 100}];
 }
+
+message OutlierEjectExtension {
+  // Name of extension monitor which caused a host ejection.
+  string monitor_name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Any extra info an extension monitor wants to add when a host is ejected.
+  string extra_info = 2;
```

Comment:
I will prefer to hide the API first. Because I found you didn't actually implement this in current implementation.

And this field is different with the name which are const value after the configuration is loaded, this field is dynamically and will bring lots of complexity.

Let's bring it back oneday if users actually want it.

Reply:
Agree. The purpose was to add extra info when criteria why monitor was tripped is not clear. I will add it when needed in the future.